### PR TITLE
Feature/cce 639 implement scripts

### PR DESCRIPTION
**Pull request by like-me1**
gulp task for compiling JS files is created. It works fine with gulp docs-js, and gulp docs-js --production and creates techne-all.js and techne-all.min.js respectively. The only one thing is that with npm start --production the environment.production variable set to false therefore I used gulp docs-js --production to test the gulp task.

### CODE REVIEW

File: docs/styleguide/templates/master.html [hunk 1]
```diff
@@ -2,12 +2,16 @@
...
   <head>
 {% include "app_meta.html" %}
-	<link rel="stylesheet" href="css/techne-{% if debug %}debug{% else %}all{% endif %}{% if env == "production" %}.min{% endif %}.css">
+	<link rel="stylesheet" href="css/techne-{% if debug %}debug{% else %}all{% endif %}{% if env == 'production' %}.min{% endif %}.css">
 
     <link rel="stylesheet" href="css/techne-styleguide.css">
 {% block css %}
 
 {% endblock %}
+      <script src="js/techne-{% if debug %}debug{% else %}all{% endif %}{% if env == 'production' %}.min{% endif %}.js"></script>
```
Comment (xak) [inline docs/styleguide/templates/master.html]: A few things ...

1. Since this JS is specific to the style guide and not the library, let's call it `techne-styleguide.js` to match the name of the styleguide CSS.
2. All of this should go at the bottom of the HTML page, before the closing `<body>` tag.
3. The debug logic can be removed since there is no debug JS.

File: ops/gulp/dev-watch.js [hunk 2]
```diff
@@ -21,6 +21,9 @@ const task = () => {
 	//update styleguide styles
 	gulp.watch([`./docs/styleguide/styles/*.scss`], ['docs-styleguide']);
 
+	//watch docs js
+	gulp.watch([`./src/scripts/*.js`,`./ops/gulp/*.js`], ['docs-js']);
```
Comment (xak) [inline ops/gulp/dev-watch.js]: No need to watch `./ops/gulp/*.js`. This would run `docs-js` when any ops JS file is modified and most of the JS files are unrelated to the site JS.

Also, the watch task here will compile the JS but will not update the browser automatically. See `dev-serve.js` where Browsersync file watching is specified to inject the JS into the page when running the local server.

File: ops/gulp/docs-js.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+const gp_concat = require('gulp-concat');
+const gp_rename = require('gulp-rename');
+const gp_uglify = require('gulp-uglify');
+const gulpif = require('gulp-if');
+const gulp_jslint = require('gulp-jslint');
+
+let environment = require('../lib/environment');
+
+const paths = {
+    src: './src/scripts',
```
Comment (xak) [inline ops/gulp/docs-js.js]: Since this JS is for the site only and not part of the library, it should be in `docs` and not `src`.

File: ops/gulp/docs-js.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    dest: './www/js'
+}
+
+let techneDestJS = environment.production ? 'techne-all.min.js' : 'techne-all.js';
+
+const task = (cb) => {
+    return gulp.src([`${paths.src}/*`])
+        .pipe(gulp_jslint())
```
Comment (xak) [inline ops/gulp/docs-js.js]: Does JS lint print errors to the console?

File: ops/gulp/docs-js.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        .pipe(gp_concat('concat.js'))
```
Comment (xak) [inline ops/gulp/docs-js.js]: I'm not sure concatenation will work here. This will take all the files in the directory and concat them in alpha order, right? This may not be a problem depending on how we organize the files but we may need to control the order of the files, for example, if we are including jQuery here it would have to be included first.

File: docs/styleguide/templates/master.html (hunk 1 above)
Comment (like-me1) [inline docs/styleguide/templates/master.html]: @xak done

File: ops/gulp/dev-watch.js (hunk 2 above)
Comment (like-me1) [inline ops/gulp/dev-watch.js]: @xak deleted ./ops/gulp/*.js from docs-js and added watcher for JS-files in dev-serve.js

File: ops/gulp/docs-js.js (hunk 3 above)
Comment (like-me1) [inline ops/gulp/docs-js.js]: done

File: ops/gulp/docs-js.js (hunk 4 above)
Comment (like-me1) [inline ops/gulp/docs-js.js]: nope, used gulp-eslint instead

File: ops/gulp/docs-js.js (hunk 5 above)
Comment (like-me1) [inline ops/gulp/docs-js.js]: when using gulp-concat files will be concatenated in the order that they are specified in the gulp.src function.If we need some custom includes we can specify it in gulp.src or use https://www.npmjs.com/package/gulp-order. But for now I'd leave it as it is. For now I can't oversee the future exceptions.

**Comment by like-me1** [comment]
@xak ready for check

**Comment by xak** [comment]
:shipit:
